cellprofiler/__main__.py

```diff
@@ -576,18 +586,24 @@ def get_batch_commands(filename):
                 print "CellProfiler -c -r -p %s -f %d -l %d" % (filename, prev + 1, off)
 
                 prev = off
-
-            return
-
-    metadata_tags = m.get_grouping_tags()
-
-    groupings = m.get_groupings(metadata_tags)
-
-    for grouping in groupings:
-        group_string = ",".join(["%s=%s" % (k, v) for k, v in grouping[0].iteritems()])
-
-        print "CellProfiler -c -r -p %s -g %s" % (filename, group_string)
-
+    else:
+        metadata_tags = m.get_grouping_tags()
+
+        if len(metadata_tags) == 1 and metadata_tags[0] == 'ImageNumber':
+            #do file numbers
+            for i in range(0, len(image_numbers), n_per_job):
+                first = image_numbers[i]
+                last  = image_numbers[min(i+n_per_job-1, len(image_numbers)-1)]
+                print "CellProfiler -c -r -p %s -f %d -l %d" % (filename, first, last)
+        else:
+            # This is the original code, doesn't work if the only metadata tag is group number
+            # not sure whether it works in other cases.
+            # Also not sure whether this code can even be reached as grouping should land us in the first "if" branch 
```

Comment:
Well, the behaviour depends on the pipeline and the generated `Batch_data.h5` file. 
I attached a simple `Batch_data.h5` file. 
Sample invocation:
```
(xprofiler) mac-almf4:CellProfiler$ python CellProfiler.py --get-batch-commands /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 1 -l 1
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 2 -l 2
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 3 -l 3
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 4 -l 4
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 5 -l 5
(xprofiler) mac-almf4:CellProfiler$ python CellProfiler.py --get-batch-commands /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 --images-per-batch 4
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 1 -l 4
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 5 -l 5
(xprofiler) mac-almf4:CellProfiler$ python CellProfiler.py --get-batch-commands /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 --images-per-batch 2
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 1 -l 2
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 3 -l 4
CellProfiler -c -r -p /Volumes/almfscreen/volker/clusteroutput/Batch_data.h5 -f 5 -l 5
```
[Batch_data.h5.zip](https://github.com/CellProfiler/CellProfiler/files/970691/Batch_data.h5.zip)